src/analysis/retail/priest/discipline/modules/spells/Expiation.tsx

```diff
@@ -0,0 +1,117 @@
+import { formatNumber } from 'common/format';
+import SPELLS from 'common/SPELLS';
+import { TALENTS_PRIEST } from 'common/TALENTS';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { Options } from 'parser/core/Module';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import ItemHealingDone from 'parser/ui/ItemHealingDone';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import { SpellLink } from 'interface';
+import AtonementAnalyzer, { AtonementAnalyzerEvent } from '../core/AtonementAnalyzer';
+import { calculateEffectiveDamage, calculateEffectiveHealing } from 'parser/core/EventCalculateLib';
+import Events, { DamageEvent } from 'parser/core/Events';
+import ItemDamageDone from 'parser/ui/ItemDamageDone';
+
+const EXPIATION_RANK_INCREASE = 0.1;
+class Expiation extends Analyzer {
+  expiationHealing = 0;
+  deathHealing = 0;
+  mindBlastHealing = 0;
+  talentRank = 0;
+  expiationIncrease = 0;
+  expiationDamage = 0;
+  bonusDamage = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS_PRIEST.EXPIATION_DISCIPLINE_TALENT.id);
+    this.addEventListener(AtonementAnalyzer.atonementEventFilter, this.onAtonement);
+
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER), this.onDamage);
```

Comment:
```suggestion
    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell([
      SPELLS.EXPIATION_DAMAGE,
      TALENTS_PRIEST.SHADOW_WORD_DEATH_TALENT,
      SPELLS.MIND_BLAST
    ]), this.onDamage);
```

you should move the damage spell filter here. probably can't do that with the custom atonement stuff, but regular damage/healing events work